pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -326,12 +345,21 @@ def _process_response(self, response: BetaMessage) -> ModelResponse:
                     )
                 )
 
+        # Map finish reason from stop_reason
+        mapped_finish = _map_anthropic_finish(getattr(response, 'stop_reason', None))
+        provider_details: dict[str, Any] | None = None
+        raw_finish: str | None = getattr(response, 'stop_reason', None)
+        if raw_finish is not None:
+            provider_details = {'finish_reason': raw_finish}
+
         return ModelResponse(
             parts=items,
             usage=_map_usage(response),
             model_name=response.model,
             provider_response_id=response.id,
             provider_name=self._provider.name,
+            finish_reason=mapped_finish,
```

Comment:
If you run `make typecheck`, you'll see an error here because we're passing a `str` to a field that requires `OtelFinishReason`